File: Parksujin/[PGS] 신고결과받기.py
# 신고 중복 제거
# reporter : 신고한 유저, 신고당한 유저 / reportee : 신고당한 유저, 신고한유저

def solution(id_list, report, k):
    answer = [] # 결과값

    reporter = {}
    reportee = {}

    for i in report:
        report_user, reported_user = i.split()
        if report_user not in reporter:
            reporter[report_user] = set()
        reporter[report_user].add(reported_user)

        if reported_user not in reportee:
            reportee[reported_user] = set()
        reportee[reported_user].add(report_user)

    answer = [0 for p in range(len(id_list))]

    for r in range(len(id_list)):
        report_user = id_list[r]
        if report_user not in reporter:
            continue

        for reported_user in reporter[report_user]:
            if len(reportee[reported_user]) >= k:
                answer[r] += 1

    return answer

File: Kojiwan/[PGS] 신고결과받기.cpp
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <algorithm>
using namespace std;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
	vector<int> answer(id_list.size(), 0);
	sort(report.begin(), report.end());
	report.erase(unique(report.begin(), report.end()), report.end());
	map<string, vector<string>> reported_list;

	for (int i = 0; i < report.size(); ++i) {
		size_t pos = report[i].find(' ');
		string reporter = report[i].substr(0, pos); // muzi
		string reported = report[i].substr(pos + 1, report[i].size() - 1); // frodo
		reported_list[reported].push_back(reporter);
	}

	for (auto iter = reported_list.begin(); iter != reported_list.end(); ++iter) {
		if (iter->second.size() < k) continue;
		for (size_t i = 0; i < iter->second.size(); ++i) {
			auto it = find(id_list.begin(), id_list.end(), iter->second[i]) - id_list.begin();
			++answer[it];
		}
	}

	for (int i = 0; i < id_list.size(); ++i) {
		cout << answer[i] << ' ';
	}


	return answer;
}